src/NHibernate/Dialect/PostgreSQL82Dialect.cs

```diff
@@ -16,6 +17,7 @@ public class PostgreSQL82Dialect : PostgreSQL81Dialect
 		public PostgreSQL82Dialect()
 		{
 			RegisterColumnType(DbType.Guid, "uuid");
+			RegisterFunction("strguid", new SQLFunctionTemplate(NHibernateUtil.String, "cast(?1 as varchar(36))"));
```

Comment:
Why having this override specific to 8.2 version? Do really previous versions need to use `nvarchar` for an alphanumeric string?
I would expect this function to work for `PostgreSQLDialect`, and if yes, it would be better to use it there.

Reply:
Yes, I made this decision where dialect declare GUID.

If I try to use GUID with  `PostgreSQLDialect` or `PostgreSQL81Dialect.` Dialect, I will get a Not Supported GUID exception